### PR TITLE
Added MaxStateSyncDataLength with hardfork

### PR DESCRIPTION
**Pull request by 0xKrishna**

### CODE REVIEW

**Review by vcastellm — APPROVED** [review]
LGTM Where `SpanOverrideBlockHeight` is defined?

**Comment by 0xKrishna** [comment]
> LGTM Where `SpanOverrideBlockHeight` is defined?

In https://github.com/maticnetwork/heimdall/pull/791/files

File: clerk/client/cli/tx.go [hunk 1]
```diff
@@ -101,9 +101,16 @@ func CreateNewStateRecord(cdc *codec.Codec) *cobra.Command {
...
-			if len(data) > helper.MaxStateSyncSize {
-				logger.Info(`Data is too large to process, Resetting to ""`, "id", recordIDStr)
-				data = hmTypes.HexToHexBytes("")
+			if cliCtx.Height > helper.SpanOverrideBlockHeight {
+				if len(data) > helper.MaxStateSyncDataLength {
+					logger.Info(`Data is too large to process, Resetting to ""`, "id", recordIDStr)
+					data = hmTypes.HexToHexBytes("")
+				}
+			} else {
+				if len(data) > helper.MaxStateSyncSize {
```
Comment (jdkanani) [inline clerk/client/cli/tx.go:110]:
```suggestion
			} else if len(data) > helper.MaxStateSyncSize {
```

File: clerk/client/cli/tx.go [hunk 2]
```diff
@@ -101,9 +101,16 @@ func CreateNewStateRecord(cdc *codec.Codec) *cobra.Command {
 				return fmt.Errorf("data should be hex string")
 			}
 
-			if len(data) > helper.MaxStateSyncSize {
-				logger.Info(`Data is too large to process, Resetting to ""`, "id", recordIDStr)
-				data = hmTypes.HexToHexBytes("")
+			if cliCtx.Height > helper.SpanOverrideBlockHeight {
+				if len(data) > helper.MaxStateSyncDataLength {
```
Comment (jdkanani) [inline clerk/client/cli/tx.go:105]:
```suggestion
			if cliCtx.Height > helper.SpanOverrideBlockHeight && len(data) > helper.MaxStateSyncDataLength {
```

File: clerk/client/rest/tx.go [hunk 3]
```diff
@@ -50,9 +50,16 @@ func newEventRecordHandler(cliCtx context.CLIContext) http.HandlerFunc {
 		// get ContractAddress
 		contractAddress := types.HexToHeimdallAddress(req.ContractAddress)
 
-		if len(types.HexToHexBytes(req.Data)) > helper.MaxStateSyncSize {
-			RestLogger.Info(`Data is too large to process, Resetting to ""`, "id", req.ID)
-			req.Data = ""
+		if cliCtx.Height > helper.SpanOverrideBlockHeight {
+			if len(types.HexToHexBytes(req.Data)) > helper.MaxStateSyncDataLength {
```
Comment (jdkanani) [inline clerk/client/rest/tx.go:54]:
```suggestion
		if cliCtx.Height > helper.SpanOverrideBlockHeight && len(types.HexToHexBytes(req.Data)) > helper.MaxStateSyncDataLength {
```

File: clerk/client/rest/tx.go [hunk 4]
```diff
@@ -50,9 +50,16 @@ func newEventRecordHandler(cliCtx context.CLIContext) http.HandlerFunc {
...
+				RestLogger.Info(`Data is too large to process, Resetting to ""`, "id", req.ID)
+				req.Data = ""
+			}
+		} else {
+			if len(types.HexToHexBytes(req.Data)) > helper.MaxStateSyncSize {
```
Comment (jdkanani) [inline clerk/client/rest/tx.go:59]:
```suggestion
		} else if len(types.HexToHexBytes(req.Data)) > helper.MaxStateSyncSize {
```

File: clerk/side_handler.go [hunk 5]
```diff
@@ -90,15 +90,25 @@ func SideHandleMsgEventRecord(ctx sdk.Context, k Keeper, msg types.MsgEventRecor
...
+			if !(len(eventLog.Data) > helper.MaxStateSyncDataLength && bytes.Equal(msg.Data, hmTypes.HexToHexBytes(""))) {
+				k.Logger(ctx).Error(
+					"Data from event does not match with Msg Data",
+					"EventData", hmTypes.BytesToHexBytes(eventLog.Data),
+					"MsgData", hmTypes.BytesToHexBytes(msg.Data),
+				)
+				return hmCommon.ErrorSideTx(k.Codespace(), common.CodeInvalidMsg)
+			}
+		} else {
+			if !(len(eventLog.Data) > helper.MaxStateSyncSize && bytes.Equal(msg.Data, hmTypes.HexToHexBytes(""))) {
```
Comment (vcastellm) [inline clerk/side_handler.go:103]: Duplicate code here?

Comment (vcastellm) [inline clerk/side_handler.go:103]: Better move the check to a method with one param?

Comment (ssandeep) [inline clerk/side_handler.go:103]: I thought we don't need this in side_hander as well?
cc/ @temaniarpit27 @jdkanani

Comment (0xKrishna) [reply clerk/side_handler.go:103]: I didn't have that context, Let me know If I should remove it from side_handler

File: bridge/setu/processor/clerk.go [hunk 6]
```diff
@@ -95,7 +95,7 @@ func (cp *ClerkProcessor) sendStateSyncedToHeimdall(eventName string, logBytes s
 			"blockNumber", vLog.BlockNumber,
 		)
 
-		if len(event.Data) > helper.MaxStateSyncSize {
+		if len(event.Data) > helper.MaxStateSyncDataLength {
```
Comment (temaniarpit27) [inline bridge/setu/processor/clerk.go:98]: check for block height

Comment (0xKrishna) [reply bridge/setu/processor/clerk.go:98]: It's in bridge, Not needed.

File: helper/config.go [hunk 7]
```diff
@@ -70,8 +70,12 @@ const (
 
 	secretFilePerm = 0600
 
+	// Legacy value - DO NOT CHANGE
 	// Maximum allowed event record data size
-	MaxStateSyncSize = 100000
+	LegacyMaxStateSyncSize = 100000
+
+	// New max state sync size after hardfork
+	MaxStateSyncSize = 30000
```
Comment (anilCSE) [inline helper/config.go:78]: Should it be 50000? Are we not using same value everywhere?

Comment (0xKrishna) [reply helper/config.go:78]: It will be 30K, I will share the details on slack.

**Review by jdkanani — APPROVED** [review]

**Pull request merged**